File: 社区相关/LeetCode/704.二分查找.cpp
/*
 * @lc app=leetcode.cn id=704 lang=cpp
 *
 * [704] 二分查找
 */

// @lc code=start
class Solution {
public:
    int search(vector<int>& nums, int target) {
        int pivot, left = 0, right = nums.size() - 1;
        while(left <= right){
            pivot = (left + right) / 2;
            if(nums[pivot]<target) left = pivot + 1;
            else if(nums[pivot] > target) right = pivot -1;
            else return pivot; 
        }
        return -1;
    }
};
// @lc code=end

File: docs/学习路线/算法/LeetCode/CodeStash/704.二分查找.py
#
# @lc app=leetcode.cn id=704 lang=python3
#
# [704] 二分查找
#

# @lc code=start
from typing import List
class Solution:
    def search(self, nums: List[int], target: int) -> int:
        if nums[0] > target or nums[-1] < target:
            return -1
        if nums[0] == target:
            return 0

        left, right = 0, len(nums) - 1
        ans = right
        while nums[ans] != target and left < right - 1:
            if nums[ans] > target:
                right = ans
                ans = (left + right) // 2
            else:
                left = ans
                ans = (left + right) // 2
        return ans if nums[ans] == target else -1
# @lc code=end
